File: lc_soln/0424_longest_repeating_character_replacement/solution.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int result = 0;
        int start = 0;
        int high = 0;
        std::vector<int> counter(26, 0);
        for (int i = 0; i < s.size(); ++i) {
            counter[s[i] - 'A']++;
            high = std::max(high, counter[s[i] - 'A']);
            while (i - start + 1 - high > k) {
                counter[s[start] - 'A']--;
                start++;
            }
            result = std::max(result, i - start + 1);
        }
        return result;
    }
};

int main() {
    return 0;
}

File: lc_soln/0424_longest_repeating_character_replacement/solution.py
from collections import defaultdict


class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        result = 0
        start = 0
        high = 0
        counter = defaultdict(int)
        for i, c in enumerate(s):
            counter[c] += 1
            high = max(high, counter[c])
            while i - start + 1 - high > k:
                counter[s[start]] -= 1
                start += 1
            result = max(result, i - start + 1)
        return result


def main(): ...


if __name__ == "__main__":
    main()
